library/core/src/ffi/c_str.rs

```diff
@@ -229,6 +232,25 @@ impl Invariant for &CStr {
     }
 }
 
+#[requires(!ptr.is_null())]
+fn is_null_terminated(ptr: *const c_char) -> bool {
+    let mut next = ptr;
+    let mut found_null = false;
+    while can_dereference(next) {
+        if unsafe { *next == 0 } {
+            found_null = true;
+            break;
+        }
+        next = next.wrapping_add(1);
+        
+        // Check distance is within isize::MAX
+        if (next.addr() - ptr.addr()) >= isize::MAX as usize {
```

Comment:
Btw, I think you can move this to be outside of the loop

Reply:
Sure, fixed